### DISCUSSION TITLE
An items.create flow should know if it's a copy or a new item

### DISCUSSION THREAD

**Discussion by notflip** [directus/directus#21656] [Feature Requests]
### Summary

I have a **Pages** model, which has a title and slug field, when the client copies that page, I need to change the slug, otherwise we have a duplicate slug. 

At this point, when using a flow, I can't know for sure if the items.create trigger is because of a copy, or because of a new item being created, hence I don't know if I need to change the title and slug to title-copy and slug-copy.

Would it be possible to add a data point to the $trigger when it's a copy?

### Motivation

Prevent duplicate slugs

### Detailed Design

A $trigger for items.create has an attribute telling us if it's a copy.

### Must Have
- A $trigger for items.create has an attribute telling us if it's a copy.

### Drawbacks

None I can think of

### Alternatives

None I can think of

### Adoption Strategy

I don't know, sorry

**Comment by paescuj** [comment]
I'm afraid that could be difficult, as a copy is just a POST request like the creation of a new item.

For your use-case: I think you could simply check in the Flow whether an item with the given slug already exists (Read Operation) and update the slug if this is the case. This should make sense for both copied and new items.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
